### PR TITLE
Add JISC Historical texts

### PR DESCRIPTION
**Pull request by adam3smith**

### CODE REVIEW

**Comment by zuphilip** [comment]
Are the automatic tests working for you? I see currently an certificate error:
```
Error: data.ukmhl.historicaltexts.jisc.ac.uk uses an invalid security certificate.
         
         The certificate is not trusted because the issuer certificate is unknown.
         The server might not be sending the appropriate intermediate certificates.
         An additional root certificate may need to be imported.
         
         Error code: SEC_ERROR_UNKNOWN_ISSUER
         )
```

File: JISC Historical Texts.js [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+// attr()/text() v2
+function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.getAttribute(attr):null;}function text(docOrElem,selector,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.textContent:null;}
+
+
+function detectWeb(doc, url) {
+
+	if (url.includes('/view?')) {
+		return "book";
+	} else if (url.includes("/results?terms")) {
```
[the remark this answers is not in the document]
Comment (adam3smith) [reply JISC Historical Texts.js]: I don't understand the question -- that page works

**Comment by dstillman** [comment]
> I see currently an certificate error

The site is misconfigured. ("Chain issues: Incomplete" on the [SSL Labs test](https://www.ssllabs.com/ssltest/analyze.html?d=data.ukmhl.historicaltexts.jisc.ac.uk&hideResults=on&latest).)

https://www.zotero.org/support/kb/incomplete_cert_chain

File: JISC Historical Texts.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (zuphilip) [reply JISC Historical Texts.js]: Okay, forget the second question, my fault here.

[the remark this answers is not in the document]
Comment (zuphilip) [reply JISC Historical Texts.js]: Maybe add a comment about why we don't test `getSearchResult` as usual and that tests for multiples will not work.

File: JISC Historical Texts.js [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+
+// attr()/text() v2
+function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.getAttribute(attr):null;}function text(docOrElem,selector,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.textContent:null;}
+
+
+function detectWeb(doc, url) {
+
+	if (url.includes('/view?')) {
+		return "book";
```
[the remark this answers is not in the document]
Comment (adam3smith) [reply JISC Historical Texts.js]: neither can I -- it looks like those are whole issues, though, so for the time being `book` would be the best item type anyway?

File: JISC Historical Texts.js [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+					}
+				],
+				"tags": [],
+				"notes": [],
+				"seeAlso": []
+			}
+		]
+	}
+]
+/** END TEST CASES **/
```
[the remark this answers is not in the document]
Comment (adam3smith) [reply JISC Historical Texts.js]: done

File: JISC Historical Texts.js [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+				pdfUrl = "/media/pdf/" + format.location;
+				//Z.debug(pdfUrl);
+				break;
+			}
+		}
+		if (pdfUrl) {
+			item.attachments.push({url:pdfUrl, title: "Historical Text PDF", mimeType: "application/pdf"});
+		}
+		item.complete();
+	});
```
[the remark this answers is not in the document]
Comment (adam3smith) [reply JISC Historical Texts.js]: done

**Comment by adam3smith** [comment]
Thanks! All good points -- I think I should have everything addressed.

File: JISC Historical Texts.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (zuphilip) [reply JISC Historical Texts.js]: Ok, possibly these are whole volumes. Then we can go with `book` for now and wait until someone with access says something different.

**Comment by zuphilip** [comment]
@adam3smith Do the automatic tests in Scaffold for you work? I couldn't run them because of the certification problem form the SSL misconfiguration which Dan confirmed.

**Comment by adam3smith** [comment]
Yeah, they work for me (Scaffold in Zotero 5.0.55 on Linux). Odd.

**Comment by zuphilip** [comment]
Odd. Anyway the code looks fine. Thank you very much!

**Comment by dstillman** [comment]
(It would work for you if your Zotero had previously loaded a site that served the same intermediate certificate properly, in which case it would be cached. It would never work in a new profile.)

**Pull request merged**